http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java

```diff
@@ -96,7 +95,9 @@ public String scheme() {
      * @see Builder#password(String)
      */
     public String username() {
-        return resolveValue(username, ProxySystemSetting.PROXY_USERNAME);
+        String httpsPort = resolveValue(null, ProxySystemSetting.HTTPS_PROXY_HOST);
+        return httpsPort != null ? resolveValue(username, ProxySystemSetting.HTTPS_PROXY_USERNAME)
+               : resolveValue(username, ProxySystemSetting.PROXY_USERNAME);
```

Comment:
Should we return usename based on the provided `scheme` instead of checking https first?

Same for other getters

Reply:
Updated to check scheme instead